backend/controllers/searchWithFilter.js

```diff
@@ -31,11 +31,11 @@ const searchFactory = (
     const pageOffset = (offset && parseInt(offset, 10)) || 0
 
     const groupCode = currentFilter.groupCode && currentFilter.groupCode[0]
-    const roleCode = (currentFilter.roleCode && currentFilter.roleCode[0]) ?? ''
+    const { roleCode } = currentFilter
 
```

Comment:
no longer need to pass a blank string when undefined now service both services deal correctly with undefined (since using stringify)